File: solutions/teque.py
import math

loop = int(input())

data = []

actions = {
    'push_back': lambda num=0: pb(num),
    'push_front':  lambda num=0: pf(num),
    'push_middle': lambda num=0: pm(num),
    'get': lambda num=0: print(data[num]) 
}

def pb(num):
    data.append(num)

def pf(num):
    data.insert(0, num)

def pm(num):
    mid = math.ceil(len(data)/2)
    data.insert(mid, num)

while loop > 0:
    val, num = input().split()
    actions[val](int(num))

    loop-=1

File: solutions/teque.cpp
#include "iostream"
#include "vector"

using namespace std;

int main() {
    
    int loop;

    vector<int> nums;
    
    cin >> loop;

    while (loop--) {
        string action;
        int num;

        cin >> action;
        cin >> num;

        if (action == "push_front") {
            nums.insert(nums.begin(), num);
        } else if (action == "push_back") {
            nums.push_back(num);
        } else if (action == "push_middle") {
            int size = (int) ((double) nums.size() / 2 + 0.5);
            nums.insert(nums.begin() + size, num);
        } else {
            cout << nums.at(num) << endl;
        }
    }

    return 0;

}